Sources/Kodable/KodableOptions+Presets.swift

```diff
@@ -25,34 +25,76 @@ public extension KodableOption {
-        .modifier(KodableModifier { (value: String?) -> String? in
+        overrideValue { (value: String?) -> String? in
             let result = value?.trimmingCharacters(in: .whitespacesAndNewlines)
             return result?.isEmpty == true ? nil : result
-        })
+        }
+    }
+}
+
+// MARK: Sort Presets
+
+public extension KodableOption where Value: Collection {
+    typealias Comparator = (Value.Element, Value.Element) -> Bool
+
+    /// Sorts the array using a `Comparator` closure that determines whether the items are in increasing order
+    static func sorted(using comparator: @escaping Comparator) -> KodableOption {
+        overrideValue { $0.sorted(by: comparator) as! Value }
+    }
+
+    /// Sorts the elements of an array ascending by a given keyPath
+    static func ascending<C: Comparable>(by keyPath: KeyPath<Value.Element, C>) -> KodableOption {
+        sorted { $0[keyPath: keyPath] < $1[keyPath: keyPath] }
+    }
+
+    /// Sorts the elements of an array descending by a given keyPath
+    static func descending<C: Comparable>(by keyPath: KeyPath<Value.Element, C>) -> KodableOption {
+        sorted { $0[keyPath: keyPath] > $1[keyPath: keyPath] }
     }
 }
 
 // MARK: Comparable Presets
 
 public extension KodableOption where Value: Comparable {
     /// Clamps the value in a range. Only applicable to types conforming to Comparable
-    static func clamping(to range: ClosedRange<Value>) -> KodableOption { .modifier(KodableModifier { $0.constrained(to: range) }) }
+    static func clamping(to range: ClosedRange<Value>) -> KodableOption { overrideValue { $0.constrained(to: range) } }
 
     /// Constrains the value inside a `range`. Only applicable to types conforming to Comparable
-    static func range(_ range: ClosedRange<Value>) -> KodableOption { .modifier(KodableModifier { $0.constrained(to: range) }) }
+    static func range(_ range: ClosedRange<Value>) -> KodableOption { overrideValue { $0.constrained(to: range) } }
 
     /// Constrains the value to a `max` value. Only applicable to types conforming to Comparable
-    static func max(_ value: Value) -> KodableOption { .modifier(KodableModifier { $0.constrained(toAtMost: value) }) }
+    static func max(_ value: Value) -> KodableOption { overrideValue { $0.constrained(toAtMost: value) } }
 
     /// Constrains the value to a `min` value. Only applicable to types conforming to Comparable
-    static func min(_ value: Value) -> KodableOption { .modifier(KodableModifier { $0.constrained(toAtLeast: value) }) }
+    static func min(_ value: Value) -> KodableOption { overrideValue { $0.constrained(toAtLeast: value) } }
+}
+
+// MARK: Collection Presets
+
+public extension KodableOption where Value: Collection, Value.Element: Comparable {
+    /// Sorts the elements of an array ascending
+    static var ascending: KodableOption { sorted { $0 < $1 } }
+    /// Sorts the elements of an array descending
+    static var descending: KodableOption { sorted { $0 > $1 } }
```

Comment:
I wonder what happens to the stability of this sorting when the array has equal elements  (i.e. $0 == $1). This is what I tried documenting in my proposal for the README where I wrote:

>…using the type's underlying comparison function.

So that it's vague enough for the user to know that this depends on their Comparable/Equatable functions 😅

Reply:
whenever the default doesn't get you where you want to be you'll be able to do your own sorting with the closure version :D